x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java

```diff
@@ -2041,41 +2041,17 @@ public void testForkFieldsWithEnrichAndLookupJoins() {
         assumeTrue("FORK available as snapshot only", EsqlCapabilities.Cap.FORK.isEnabled());
         assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V12.isEnabled());
 
-        assertFieldNames(
-            """
-                FROM test
-                | KEEP a, b, abc, def, z, xyz
-                | ENRICH enrich_policy ON abc
-                | EVAL b = a + 100
-                | LOOKUP JOIN my_lookup_index ON def
-                | FORK (WHERE c > 1 AND a < 10000 | EVAL d = a + 500)
-                       (STATS x = count(*), y=min(z))
-                | LOOKUP JOIN my_lookup_index ON xyz
-                | WHERE x > y OR _fork == "fork1"
-                """,
-            Set.of(
-                "x",
-                "y",
-                "_fork",
-                "a",
-                "c",
-                "abc",
-                "b",
-                "def",
-                "z",
-                "xyz",
-                "def.*",
-                "_fork.*",
-                "y.*",
-                "x.*",
-                "xyz.*",
-                "z.*",
-                "abc.*",
-                "a.*",
-                "c.*",
-                "b.*"
-            )
-        );
+        assertFieldNames("""
+            FROM test
+            | KEEP a, b, abc, def, z, xyz
+            | ENRICH enrich_policy ON abc
+            | EVAL b = a + 100
+            | LOOKUP JOIN my_lookup_index ON def
+            | FORK (WHERE c > 1 AND a < 10000 | EVAL d = a + 500)
+                   (STATS x = count(*), y=min(z))
+            | LOOKUP JOIN my_lookup_index ON xyz
+            | WHERE x > y OR _fork == "fork1"
+            """, Set.of("a", "c", "abc", "b", "def", "z", "xyz", "def.*", "xyz.*", "z.*", "abc.*", "a.*", "c.*", "b.*"));
```

Comment:
this actually got better - because we stopped asking field caps for `_fork` and `b` (which was the result of an eval).